### PR TITLE
HW #2 (Linter & Code review)

### PR DESCRIPTION
**Pull request by eluyema**
- Update eslint config
- Add  workflow that checks for prettier and eslint errors
- Fix new lint errors

### CODE REVIEW

**Comment by cholombytko** [comment]
I suggest that you could improve the subscription request workflow by adding request body validation. Ensuring that email field is present and properly formatted will help prevent potential issues and enhance the overall reliability of the application.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+
+end_of_line = lf
+insert_final_newline = true
```
Comment (Larisa-Antsifrova) [inline .editorconfig:10]: Oh, nice) It looks like those files belong to the config of IDE) Should they be in the final PR?

File: .eslintrc.js [hunk 2]
```diff
@@ -1,25 +1,66 @@
...
+      typescript: {
+        project: './tsconfig.json',
+      },
+    },
+  },
+  rules: {
+    '@typescript-eslint/interface-name-prefix': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
```
Comment (Larisa-Antsifrova) [inline .eslintrc.js]: Just curious: what is the motivation behind the choice of the turned off rules?)

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.11.0'
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+      - name: Check prettier
+        run: pnpm format:check
+      - name: Check eslint
+        run: pnpm lint:check
```
Comment (Larisa-Antsifrova) [inline .github/workflows/ci-cd.yml:25]: That is nice that you've covered the extra task! Good job 😎

File: README.md [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
-1. You need to create .env file (look at example .env-sample)
-2. If you use .env-sample, than you need to set up SMPT env variables and API_KEY which you can take free from [openexchangerates.org](https://openexchangerates.org/)
-3. Run docker compose file with commands "docker-compose up"
-
-This application have unit tests, you can ran it with command:
-
-```bash
-$ npx pnpm run test
-```
+**This project use DDD architecture, node.js, nest.js, prisma, nodemailer.**
```
Comment (Larisa-Antsifrova) [inline README.md:1]:
```suggestion
**This project uses DDD architecture, node.js, nest.js, prisma, nodemailer.**
```

File: README.md [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
+
+For mail template it use handlebars templates.
```
Comment (Larisa-Antsifrova) [inline README.md]:
```suggestion
For mail template it uses handlebars templates.
```

File: README.md [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
+
+How to start application?
+
+1. You need to create .env file (look at example .env-sample)
+2. If you use .env-sample, than you need to set up SMPT env variables and API_KEY which you can take free from [openexchangerates.org](https://openexchangerates.org/)
+3. Run docker compose file with commands "docker-compose up"
+
+This application have unit tests, you can ran it with command:
```
Comment (Larisa-Antsifrova) [inline README.md]:
```suggestion
This application has unit tests, you can run it with command:
```

File: vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "editor.formatOnSave": true,
+  "eslint.validate": ["typescript"],
+  "editor.codeActionsOnSave": {
+    "source.fixAll": true
+  }
+}
```
Comment (Larisa-Antsifrova) [inline vscode/settings.json]: This file also looks like the one for personal use :) However, you might come from a team with another code style agreements. I am curious to hear, if that is so :)

File: src/modules/mailer/infrastructure/templates/exchange-rate.hbs [hunk 8]
```diff
@@ -1,38 +1,18 @@
...
+      <p class='mail-text'>The exchange rate from USD to UAH is
+        <strong class='exchange-rate-price'>{{rate}}</strong></p>
+      <div class='date-block'><strong class='date-text'>{{date}}</strong></div>
+    </div>
+    <style>
+      .content-block { max-width: 600px !important; background-color: #74a9ff;
+      padding: 24px; } .mail-text { font-size: 18px; color: #fff; padding: 0; }
+      .date-block { margin-top: 20px; } .date-text { font-size: 16px; color:
+      #fff; } .exchange-rate-price { font-size: 20px; color: #fdff7b; }
+    </style>
```
Comment (Larisa-Antsifrova) [inline src/modules/mailer/infrastructure/templates/exchange-rate.hbs:16]: Non-blocking: wow! That's an interesting formatting outcome :)

**Comment by cholombytko** [comment]
I suggest that you could improve readability by inverting if. 
https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-eluyema/blob/373d3a0164c140f5fc7ace41a294e9699bc37d51/src/modules/subscription/domain/services/subscription.service.ts#L16-L20

**Comment by cholombytko** [comment]
That's all i have. You did very good architecture, gj!

File: .editorconfig (hunk 1 above)
Comment (eluyema) [reply .editorconfig:10]: Yes, it should. editorconfig standardizes the editor configuration for any (most famous) IDE.

File: .eslintrc.js (hunk 2 above)
Comment (eluyema) [reply .eslintrc.js]: For most of them, I have already forgotten, so I will remove this rules. These turned-off rules no longer throw errors, so I think they're not needed anymore. I had some problems with linters that changed the code into some ugly stuff, so I decided to turn off these rules to keep the code looking clean

File: vscode/settings.json (hunk 7 above)
Comment (eluyema) [reply vscode/settings.json]: This file is not some team's code style agreements. I will move it to gitignore

**Pull request merged**